extensions/applicationinsights-dependencies-js/src/ajaxRecord.ts

```diff
@@ -1,83 +1,361 @@
+    *  |-------------------networkConnect----------------|
+    *  |                                                   |---------sentRequest--------|
+    *  |------------------------------------perfTotal-----------------------------------|
+    */
+
+    let resourceEntry = ajaxData.perfTiming;
+    let props = dependency[strProperties] || {};
+    let propsSet = 0;
+    let _isString = CoreUtils.isString;
+    let strName = "name";
+    let strStart = "Start";
+    let strEnd = "End";
+    let strDomainLookup = "domainLookup";
+    let strConnect = "connect";
+    let strRedirect = "redirect";
+    let strRequest = "request";
+    let strResponse = "response";
+    let strDuration = "duration";
+    let strStartTime = "startTime";
+    let strDomainLookupStart = strDomainLookup + strStart;
+    let strDomainLookupEnd = strDomainLookup + strEnd;
+    let strConnectStart = strConnect + strStart;
+    let strConnectEnd = strConnect + strEnd;
+    let strRequestStart = strRequest + strStart;
+    let strRequestEnd = strRequest + strEnd;
+    let strResponseStart = strResponse + strStart;
+    let strResponseEnd = strResponse + strEnd;
+    let strRedirectStart = strRedirect + strStart;
+    let strRedirectEnd = strRedirect = strEnd;
+    
+    let strTransferSize = "transferSize";
+    let strEncodedBodySize = "encodedBodySize";
+    let strDecodedBodySize = "decodedBodySize";
+    let strServerTiming = "serverTiming"
+    
+    if (resourceEntry) {
+        // redirect
+        propsSet |= _setPerfDuration(props, strRedirect, resourceEntry, strRedirectStart, strRedirectEnd);
+
+        // domainLookup
+        propsSet |= _setPerfDuration(props, strDomainLookup, resourceEntry, strDomainLookupStart, strDomainLookupEnd);
+
+        // connect
+        propsSet |= _setPerfDuration(props, strConnect, resourceEntry, strConnectStart, strConnectEnd);
+
+        // request
+        propsSet |= _setPerfDuration(props, strRequest, resourceEntry, strRequestStart, strRequestEnd);
+
+        // response
+        propsSet |= _setPerfDuration(props, strResponse, resourceEntry, strResponseStart, strResponseEnd);
+
+        // Network connection time
+        propsSet |= _setPerfDuration(props, "networkConnect", resourceEntry, strStartTime, strConnectEnd);
+
+        // Sent Request
+        propsSet |= _setPerfDuration(props, "sentRequest", resourceEntry, strRequestStart, strResponseEnd);
+
+        // PerfTotal / Duration
+        let duration = resourceEntry[strDuration];
+        if (!duration) {
+            duration = _calcPerfDuration(resourceEntry, strStartTime, strResponseEnd) || 0;
+        }
+
+        propsSet |= _setPerfValue(props, strDuration, duration);
+        propsSet |= _setPerfValue(props, "perfTotal", duration);
+
+        var serverTiming = resourceEntry[strServerTiming];
+        if (serverTiming) {
+            let server = {};
+            _arrForEach(serverTiming, (value, idx) => {
+                let name = value[strName] || "" + idx;
```

Comment:
Should add some additional checks / conversion to make sure the name can be validly set as a key to avoid exceptions.